Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -3446,50 +3451,133 @@ def panorama_delete_url_filter_command(url_filter_name: str):
 ''' Security Rules Managing '''
 
 
+def xml_get(elem: dict | Any, key: Any, subkeys=('member', '#text', '@name')):
+    if type(elem) is not dict:
+        return elem
+
+    value = elem.get(key)
+
+    if type(value) is dict:
+        return next(
+            (xml_get(value, sub)
+             for sub in subkeys
+             if sub in value),
+            None
+        ) or next(
+            (xml_get(value, k)
+             for k in value.keys()
+             if not str(k).startswith('@')),
+            None
+        )
+    if type(value) is list:
+        return [
+            xml_get({0: item}, 0)
+            for item in value
+        ]
+
+    return value
```

Comment:
why do you need all of this?

you can use the `parse_pan_os_un_committed_data` which does all of this for you.

what's the purpose of this function? if i understand its to get data from un-committed data?

Reply:
@GuyAfik does this seem like a good function to use in panos? Is the complexity warranted?
If yes lets write very clear docstrings and tests

Reply:
from what i understand, the purpose here is to extract nested fields from the response, there are advantages which makes it easier to handle nested values recursively, however IMO, it makes the readability a bit harder and we want it to be as easy as possible to understand in case we would have to deal with bugs in the future.

from my point of view i don't see any harm as long as we can generalize it to other cases / other XML api's which might use it. 

We should definitely add very good unit-tests here to cover as many cases as possible.

Reply:
lets try to write this function clearer as well. it shouldnt be recursive

Reply:
ill work on making the function better